src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,393 @@
+    #in ISCE2 code: if float(self.IPFversion) < 2.90:
+    # REF: .../isce2/components/isceobj/Sensor/GRD/Sentinel1.py
+
+    rg_list_azimuth_time:np.ndarray
+    rg_list_line:list
+    rg_list_pixel:list
+    rg_list_noise_range_lut:list
+    az_first_azimuth_line:int
+    az_first_range_sample:int
+    az_last_azimuth_line:int
+    az_last_range_sample:int
+    az_line:np.ndarray
+    az_noise_azimuth_lut:np.ndarray
+
+    @classmethod
+    def from_et(cls,et_in,ipf_version=3.10):
+        '''Extracts list of noise information from etree'''
+        if et_in is not None:
+            cls.xml_et=et_in
+
+        if ipf_version<2.90: #legacy SAFE data
+            cls.rg_list_azimuth_time=cls._parse_vectorlist('noiseVectorList','azimuthTime','datetime')
+            cls.rg_list_line=cls._parse_vectorlist('noiseVectorList','line','scalar_int')
+            cls.rg_list_pixel=cls._parse_vectorlist('noiseVectorList','pixel','vector_int')
+            cls.rg_list_noise_range_lut=cls._parse_vectorlist('noiseVectorList','noiseLut','vector_float')
+            cls.az_first_azimuth_line=None
+            cls.az_first_range_sample=None
+            cls.az_last_azimuth_line=None
+            cls.az_last_range_sample=None
+            cls.az_line=None
+            cls.az_noise_azimuth_lut=None
+
+        else:
+            cls.rg_list_azimuth_time=cls._parse_vectorlist('noiseRangeVectorList','azimuthTime','datetime')
+            cls.rg_list_line=cls._parse_vectorlist('noiseRangeVectorList','line','scalar_int')
+            cls.rg_list_pixel=cls._parse_vectorlist('noiseRangeVectorList','pixel','vector_int')
+            cls.rg_list_noise_range_lut=cls._parse_vectorlist('noiseRangeVectorList','noiseRangeLut','vector_float')
+            cls.az_first_azimuth_line=cls._parse_vectorlist('noiseAzimuthVectorList','firstAzimuthLine','scalar_int')[0]
+            cls.az_first_range_sample=cls._parse_vectorlist('noiseAzimuthVectorList','firstRangeSample','scalar_int')[0]
+            cls.az_last_azimuth_line=cls._parse_vectorlist('noiseAzimuthVectorList','lastAzimuthLine','scalar_int')[0]
+            cls.az_last_range_sample=cls._parse_vectorlist('noiseAzimuthVectorList','lastRangeSample','scalar_int')[0]
+            cls.az_line=cls._parse_vectorlist('noiseAzimuthVectorList','line','vector_int')[0]
+            cls.az_noise_azimuth_lut=cls._parse_vectorlist('noiseAzimuthVectorList','noiseAzimuthLut','vector_float')[0]
+
+        return cls
+
+
+@dataclass
+class ProductAnnotation(AnnotationBase):
+    '''For L1 SLC product annotation file. For EAP correction.'''
+    image_information_slant_range_time:float
+    #elevation_angle:
+    antenna_pattern_azimuth_time:list
+    antenna_pattern_slant_range_time:list
+    antenna_pattern_elevation_angle:list
+    antenna_pattern_elevation_pattern:list
+
+    ascending_node_time:datetime
+    number_of_samples:int
+    range_sampling_rate:float
```

Comment:
Space after each `:` for readability and consistency [standard Python](https://docs.python.org/3/library/dataclasses.html).

Reply:
Spaces are added after `:` for other dataclasses.